Assets/Scripts/Player/PlayerController.cs

```diff
@@ -73,14 +74,20 @@ public class PlayerController : MonoBehaviour {
 			scale.x = -1;
 		transform.localScale = scale;
 
-		if (rb2D.velocity.x == 0)
+		if (Mathf.Abs(rb2D.velocity.x) < 0.01f)
 			velocity.x = 0;
 
+		// Cancels jumps if head hits roof
+		if (Mathf.Abs(rb2D.velocity.y) < 0.01f && !isGrounded) {
+			doJump = false;
+			doAirJump = false;
+		}
+
 		if (xInput != 0) {
 			float newVelocityX = velocity.x + xInput * acceleration * Time.deltaTime;
 			velocity.x = Mathf.Clamp(newVelocityX, -maxSpeed, maxSpeed);
 		}
-		else if (allowControls || isGrounded) {
+		else if ((allowControls || isGrounded) && velocity.x > 0) {
```

Comment:
Now there is no deacceleration when moving to the left.

Reply:
Good eye, will fix.